nixos/modules/services/networking/tailscale.nix

```diff
@@ -61,12 +61,21 @@ in {
     };
 
     extraUpFlags = mkOption {
-      description = "Extra flags to pass to {command}`tailscale up`.";
+      description = ''
+        Extra flags to pass to {command}`tailscale up`. Only applied if `authKeyFile` is specified.";
+      '';
       type = types.listOf types.str;
       default = [];
       example = ["--ssh"];
     };
 
+    extraSetFlags = mkOption {
+      description = "Extra flags to pass to {command}`tailscale set`.";
+      type = types.listOf types.str;
+      default = [];
+      example = ["--advertise-exit-node"];
+    };
```

Comment:
I am not convinced that we need the second option. If you once logged into tailscale via headscale, the up command also works as expected.

Reply:
I've took a 2nd look at the condition inside of extraUpFlags and I think this is correct now.